### PR TITLE
Fixes #4

### LINKED ISSUE #4: cue content causing 'ParserError', due to empty lines
I have extracted a subtitle track, as webvtt, from an MPEG4 file (from Contour+2 camera), using ffmpeg. Due to empty lines in the text of 'cue' block this is causing the following exception:

```
.../node_modules/node-webvtt/lib/parser.js:65
    throw new ParserError(msg);
    ^
Error: Cue identifier needs to be followed by timestamp (cue #1)
```

With a sample data from the webvtt file being:

```
WEBVTT

00:00.000 --> 00:01.000
FW version:1700 V2.1.29
FW name: ContourPlus2

UPDATE:N
UPDATE_FW:N

    SWITCH_1
1RES:D
1BR:H
1EV:0
1SHRP:3
1AE:C
1CTST:62
1MIC:45
1EXT_MIC:30
1SILENT:1
1LSR:1
1LED:0
1GPS_PWR:0
1GPS_REC:1
1AWB:0

    SWITCH_2
2RES:A
2BR:H
2EV:0
2SHRP:3
2AE:C
2CTST:62
2MIC:45
2EXT_MIC:30
2SILENT:0
2LSR:0
2LED:0
2GPS_PWR:0
2GPS_REC:1
2AWB:0
```

Is there any way we can get node-webvtt to deal with this? The current thought being to require all cue block be empty + properly formatted start/end time line and then treating the rest as cue content? I haven't read enough of the WebVTT spec to know whether this would cause issue.

### PR DESCRIPTION
**Pull request by ajmas**
Created fix to address issue #4. This makes the parser a little more lenient, such that cues are considered to start with an empty line, followed by start --> end line.

### CODE REVIEW

**Comment by ajmas** [comment]
Didn't run the test cases. Running into failing issues now that I am.

**Comment by ajmas** [comment]
Another small issue. While this does pass the test cases, it fails in my actual scenario. Changing my regex to be `/^\d+$/.test(lines[0])` fixes the issue in my scenarios, but then causes the test cases to fail. Any suggestions?

**Comment by ajmas** [comment]
Ok, this looks good now. Please let me know if this parses muster.

File: lib/parser.js [hunk 1]
```diff
@@ -7,9 +7,15 @@ function ParserError (message, error) {
 ParserError.prototype = Object.create(Error.prototype);
 
 const TIMESTAMP_REGEXP = /([0-9]{2})?:?([0-9]{2}):([0-9]{2}\.[0-9]{3})/;
+const TIME_SEGMENT_REGEXP = /((\d+:)?\d\d\:\d\d[.,]\d\d\d)/;
+const CUE_HEADER_REGEXP = new RegExp(`${TIME_SEGMENT_REGEXP.source}`
+     + `\\s-->\\s`
+     + `${TIME_SEGMENT_REGEXP.source}(\s.*)?`);
```
Comment (osk) [inline lib/parser.js:13]: Template strings can include newlines, so concatenation is not needed here:

```
`${TIME_SEGMENT_REGEXP.source}
\\s-->\\s
${TIME_SEGMENT_REGEXP.source}(\s.*)?`
```

Comment (osk) [inline lib/parser.js:13]: Also not really hot for creating such a complex regex for this

File: lib/parser.js [hunk 2]
```diff
@@ -36,6 +42,18 @@ function parse (input) {
     throw new ParserError('No blank line after signature');
   }
 
+  for (i = 0; i < parts.length; i++) {
```
Comment (osk) [inline lib/parser.js:45]: move `let` to here from line 17

Comment (osk) [inline lib/parser.js:45]: Would prefer using `forEach` here

**Comment by osk** [comment]
Nice work! Some comments.
Adding tests for the headers would also be awesome 😀

**Comment by kthelgason** [comment]
I'm -1 on this. Our parser conforms to the WebVTT spec as-is. The files this change is designed to parse are not valid. 
@osk what do you think? Personally I'd like to declare this a `#wontfix`

**Pull request closed without merging**